### ISSUE TITLE
Unexpected reportIncompatibleMethodOverride errors using multiple inheritance with generic bases

### ISSUE THREAD

**Issue by joell** [microsoft/pyright#2894]
**Describe the bug**
It seems that when you construct a multiple-inheritance hierarchy of generic classes in certain ways and instantiate their generic type parameter in your concrete subclass definition that Pyright will fail to fully resolve the type parameter of the generic base.  This results in strange `reportIncompatibleMethodOverride` errors.

**To Reproduce**
Typecheck this file:

```python
# pyright: strict
from typing import Generic, Iterable, TypeVar

T = TypeVar('T')

class Base(Generic[T]):
    def __init__(self, thing: T):
        self.thing = thing

    def dostuff(self) -> Iterable[T]:
        return [self.thing] * 3

class Derived(Base[T]):
    def dostuff(self) -> Iterable[T]:
        return [*super().dostuff(), *([self.thing] * 2)]

class Extra(Base[T]):
    def extra(self) -> int:
        return 42

class ThisFails(Extra[int], Derived[int]):
    pass

class EvenThisFails(Extra[int], Base[int]):
    pass

class ButThisSucceeds(Derived[int], Base[int]):
    pass
```

**Expected behavior**
The example file should typecheck cleanly.

**Observed behavior**
I get the following error:
```
test.py:21:7 - error: Base classes for class "ThisFails" define method "dostuff" in incompatible way
  Return type mismatch: base method returns type "Iterable[int]", override returns type "Iterable[T@Base]"
    TypeVar "_T_co@Iterable" is covariant
      "object*" is incompatible with "int" (reportIncompatibleMethodOverride)
test.py:24:7 - error: Base classes for class "EvenThisFails" define method "dostuff" in incompatible way
  Return type mismatch: base method returns type "Iterable[int]", override returns type "Iterable[T@Base]"
    TypeVar "_T_co@Iterable" is covariant
      "object*" is incompatible with "int" (reportIncompatibleMethodOverride)
```

**VS Code extension or command-line**
I'm running Pyright 1.1.210 directly on the command-line.
My OS is Arch Linux with Python 3.10 and Linux kernel 5.10.90.

**Comment by erictraut** [comment]
Thanks for the excellent bug report.

One of the code paths was missing a class specialization step, which resulted in this error. This will be fixed in the next release. I've also added a test case based on your above sample.

**Comment by joell** [comment]
You're welcome for the bug report.  Thank you for your assistance and incredibly fast response.  You and your team's work on this project for the community is immensely appreciated!

**Comment by erictraut** [comment]
This is included in pyright 1.1.212, which I just published. It will also be included in the next release of pylance.
